library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/Ac3Reader.java

```diff
@@ -188,18 +192,36 @@ private boolean skipToNextSync(ParsableByteArray pesBuffer) {
   private void parseHeader() {
     headerScratchBits.setPosition(0);
     Ac3Util.Ac3SyncFrameInfo frameInfo = Ac3Util.parseAc3SyncframeInfo(headerScratchBits);
-    if (format == null || frameInfo.channelCount != format.channelCount
-        || frameInfo.sampleRate != format.sampleRate
-        || frameInfo.mimeType != format.sampleMimeType) {
+    if ((format == null || frameInfo.sampleRate != format.sampleRate
+        || frameInfo.mimeType != format.sampleMimeType)
```

Comment:
I asked the original author today. He said he didn't think the channel count should be changed in EAC3 stream. (Although it can be changed in spec. But the real stream in market doesn't change it.) Shall we support this case? If yes, I have below updated code:
`    Ac3Util.Ac3SyncFrameInfo frameInfo = Ac3Util.parseAc3SyncframeInfo(headerScratchBits);
    int formatChannelCount = frameInfo.channelCount;
    if (format != null && frameInfo.streamType == Ac3Util.STREAM_TYPE_TYPE1
        && lastStreamType == Ac3Util.STREAM_TYPE_TYPE0
        && lastChannelCount == 6 && formatChannelCount == 4) {
      formatChannelCount = 8;
    }    
    if (format != null && frameInfo.streamType == Ac3Util.STREAM_TYPE_TYPE0
        && lastStreamType == Ac3Util.STREAM_TYPE_TYPE1    
        && lastChannelCount == 4 && formatChannelCount == 6) {
      formatChannelCount = 8;
    }    
    if (format == null || formatChannelCount != format.channelCount
       || frameInfo.sampleRate != format.sampleRate
       || frameInfo.mimeType != format.sampleMimeType) {
      format = Format.createAudioSampleFormat(trackFormatId, frameInfo.mimeType, null,
          Format.NO_VALUE, Format.NO_VALUE, formatChannelCount, frameInfo.sampleRate, null,
          null, 0, language);
      output.format(format);
    }

    lastStreamType = frameInfo.streamType;
    lastChannelCount = frameInfo.channelCount; `